lib/NetworkClient.js

```diff
@@ -19,7 +19,11 @@ function readRequest(requestData)
             return;
         }
         if (!error && response.statusCode === 200) {
-            d.resolve(body);
+            if (!requestData.method || requestData.method != 'HEAD') {
```

Comment:
don't like this much.... 
maybe add resolveWithFullResponse to the requestData and if this true then return
d.resolve ({body,response.headers})
(so we can reuse this)